### ISSUE TITLE
Modal not firing events when closed via backdrop click

### ISSUE THREAD

**Issue by jakob-o** [zurb/foundation-apps#641]
As the title states the foundation API does publish events when opening the modal and closing it via a close button but not when closing the modal via click on the backdrop. I created a minimal sample project to check it out here: https://github.com/jakob-o/modal-api-test

The modal code is copied from here http://foundation.zurb.com/apps/docs/#!/modal
```html
<a zf-open="basicModal" class="button">Open modal</a>
<div zf-modal="" id="basicModal">
  <a zf-close="" class="close-button">×</a>
  <h3>Back in my days...</h3>
  <p>There were 3 Star Wars movies and Disney only made cartoons.</p>
</div>
```
The controller code / API usage basically is this:
```javascript
function MainController($log, FoundationApi) {
    FoundationApi.subscribe('basicModal', function(event) {
        $log.info('Modal ' + event);
    });
}
```

If this is intended behaviour, is there a way to test for the current state (open / closed)?
Thanks for the great product and investigating this issue
Regards
Jakob

Edit: Is there btw. any documentation about the API besides this issue https://github.com/zurb/foundation-apps/issues/373?

**Comment by gakimball** [comment]
There is documentation on our modules [here](http://foundation.zurb.com/apps/docs/#!/angular-modules), but we could be doing a better job. There were talks to write more wrapper methods to make the API a bit less terse (see #390), but there's nobody working on it right now.

**Comment by gakimball** [comment]
As for the event firing, a modal should always fire an event when it closes, no matter how the close happened. That makes sense.

**Comment by jakob-o** [comment]
FYI: In the current master 36383ace3ae20c5e447da75005ea5a09a69ffa5a it looks like the 'hide' event changed to 'close' and is reliably firing.

**Comment by gakimball** [comment]
That's for popups, not modals. But your issue is fixed in master?

**Comment by jakob-o** [comment]
I didn't mean the commit itself, rather wanted to point to the current head in master. It is fixed there. You can easily test it by changing the dependency in the bower.json of the test project (see the first post) to:
```javascript
  "dependencies": {
    "foundation-apps": "https://github.com/zurb/foundation-apps.git#36383ace3ae20c5e447da75005ea5a09a69ffa5a"
  }
```
And double checking it seems to have been a "close" event before (could have sworn it issued a "hide"). So in a nutshell, with 1.1.0 close on button issues a 'close' event, click on backdrop does not. In current master both seem to work.

**Comment by jakob-o** [comment]
I guess this is the commit, which fixed it 8b0b5bbf5637192dcbdaa542dbace0dad3b03473.

**Comment by gakimball** [comment]
Gotcha. Looks like this is fixed, then.
